yoti_python_sdk/profile.py

```diff
@@ -14,10 +15,15 @@ def __init__(self, profile_attributes):
 
         if profile_attributes:
             for field in profile_attributes:
-                value = Protobuf().value_based_on_content_type(
-                    field.value,
-                    field.content_type
-                )
+                try:
+                    value = Protobuf().value_based_on_content_type(
+                        field.value,
+                        field.content_type
+                    )
+                except Exception as exc:

```

Comment:
This will catch `raise TypeError('Wrong content type')` which means attributes with the `UNDEFINED` content type will be returned as an Attribute object with a `None` value - is this a breaking change?

We now return attributes with `UNDEFINED` content type as `NULL` in the PHP SDK - should this change in behaviour remain as it is until we standardise the handling on the `UNDEFINED` content type.

Perhaps creation of `Attribute(field.name, value, anchors)` should be moved into the try block? Or alternatively, we change the behaviour in the PHP SDK.

Reply:
> This will catch `raise TypeError('Wrong content type')` which means attributes with the `UNDEFINED` content type will be returned as an Attribute object with a `None` value - is this a breaking change?

I think we have to treat this in a similar way to PHP - assuming we never had a protobuf type of undefined sent through from Core, I wouldn't count it as a breaking change if we are changing the behaviour of undefined content types. 

> We now return attributes with `UNDEFINED` content type as `NULL` in the PHP SDK - should this change in behaviour remain as it is until we standardise the handling of the `UNDEFINED` content type?

We can discuss this on Monday, I won't merge this PR until we have agreed on a common approach. 

> Perhaps creation of `Attribute(field.name, value, anchors)` should be moved into the try block? Or alternatively, we change the behaviour in the PHP SDK.

I did have it as that originally, but I thought I didn't want failing to parse the attribute affecting whether the anchors were parsed. On second thoughts, I think I should add a different try/catch block to parsing the anchors - then one of anchors/values failing won't affect the parsing of the other one. What do you think?

Reply:
Something like this?

```python
try:
    value = Protobuf().value_based_on_content_type(
      field.value,
      field.content_type
    )
    anchors = Anchor().parse_anchors(field.anchors)
    attr = Attribute(field.name, value, anchors)
except Exception as exc:
    error = 'Error parsing profile attribute: "{0}"'.format(field.name)
    logging.warning('error: {0}, exception: {1} - {2}'.format(error, type(exc).__name__, exc))
    attr = None

self.attributes[field.name] = attr
```

This should then return `None` for attributes that have `UNDEFINED` content type

Reply:
Or separate try/catch like this?

```python
try:
    anchors = Anchor().parse_anchors(field.anchors)
except Exception as exc:
    error = 'Error parsing anchors: "{0}"'.format(field.name)
    logging.warning('error: {0}, exception: {1} - {2}'.format(error, type(exc).__name__, exc))
    anchors = None

try:
    value = Protobuf().value_based_on_content_type(
      field.value,
      field.content_type
    )
    attr = Attribute(field.name, value, anchors)
except Exception as exc:
    error = 'Error parsing profile attribute: "{0}"'.format(field.name)
    logging.warning('error: {0}, exception: {1} - {2}'.format(error, type(exc).__name__, exc))
    attr = None

self.attributes[field.name] = attr
```

Reply:
Have changed it to be like your first suggestion, and not returning an attribute, in fitting what Java does.